src/mock_alchemy/mocking.py

```diff
@@ -685,3 +703,100 @@ def _mutate_data(self, *args: Any, **kwargs: Any) -> Optional[int]:
                     temp_mock_data.append((calls, result))
             self._mock_data = temp_mock_data
             return num_deleted
+        # execute case
+        else:
```

Comment:
Is there an opportunity here to break to down this new logic into a new function?